### PR TITLE
PR - Code review - Transactions and aggregators

### PR DESCRIPTION
**Pull request by wickathou**
For this PR, additional data and an extra column have been added to the database. Also, some aggregator queries are run to find details in the data.

**Features:**
- Added new data added by updating entries in the table
- Created a new 'species' column on the table 'animals'
- Special queries have been created to extract information from the data stored

Screenshots:
- Inside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
[Task 0](https://github.com/wickathou/vet-clinic-sql-practice/blob/transactions/screenshots/day2/queries-day2-0.png)

- Inside a transaction:
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Commit the transaction.
Verify that change was made and persists after commit.
Take a screenshot of the results of your actions.
[Task 1](https://github.com/wickathou/vet-clinic-sql-practice/blob/transactions/screenshots/day2/queries-day2-1.png)

- Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)
Take a screenshot of the results of your actions.
[Task 2](https://github.com/wickathou/vet-clinic-sql-practice/blob/transactions/screenshots/day2/queries-day2-2.png)

- Inside a transaction:
Delete all animals born after Jan 1st, 2022.
Create a savepoint for the transaction.
Update all animals' weight to be their weight multiplied by -1.
Rollback to the savepoint
Update all animals' weights that are negative to be their weight multiplied by -1.
Commit transaction
Take a screenshot of the results of your actions.
[Task 3](https://github.com/wickathou/vet-clinic-sql-practice/blob/transactions/screenshots/day2/queries-day2-3.png)

- Write queries to answer the following questions:
How many animals are there?
How many animals have never tried to escape?
What is the average weight of animals?
Who escapes the most, neutered or not neutered animals?
What is the minimum and maximum weight of each type of animal?
What is the average number of escape attempts per animal type of those born between 1990 and 2000?
Take a screenshot of the results of your queries.
[Task 4](https://github.com/wickathou/vet-clinic-sql-practice/blob/transactions/screenshots/day2/queries-day2-4.png)

### CODE REVIEW

File: data.sql [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+
 insert into animals(name,date_of_birth, weight_kg,neutered,escape_attempts) values ('Agumon','2020-02-03', 10.23,true,0);
 insert into animals(name,date_of_birth, weight_kg,neutered,escape_attempts) values 
   ('Gabumon','2018-11-15', 8,true,2),
   ('Pikachu','2021-01-07',15.04,false,1),
   ('Devimon','2017-05-12', 11,true,5);
+
+-- Day 2
+
+alter table animals add column species varchar(100) ;
```
Comment (bobb-Rob) [inline data.sql]: - [x] **_Everything that creates or changes the table structure should be saved in the schema.sql file._** This command here adds a specie column to the `animals` table, and this in effect changes the table, therefore, it should be placed in the `schema.sql` file. Kindly fix this

File: queries.sql [hunk 2]
```diff
@@ -6,3 +6,64 @@ select (name,escape_attempts) from animals where weight_kg > 10.5;
 select * from animals where neutered is true;
 select * from animals where name != 'Gabumon';
 select * from animals where weight_kg >= 10.4 and weight_kg <= 17.3;
+
+-- Day 2
+
+begin;
+update animals set species = 'unspecified';
+select * from animals;
+rollback;
```
Comment (bobb-Rob) [inline queries.sql:15]: - [x] After the rollback on line 15, you should add a query to verify that the change was undone.

File: queries.sql [hunk 3]
```diff
@@ -6,3 +6,64 @@ select (name,escape_attempts) from animals where weight_kg > 10.5;
...
+
+begin;
+update animals set species='digimon' where name like '%mon';
+update animals set species='pokemon' where species is null;
+select * from animals;
+commit;
```
Comment (bobb-Rob) [inline queries.sql:22]: - [x] You should have a query that verify that change persists after commit.

```sql
SELECT species from animals;
```

File: queries.sql [hunk 4]
```diff
@@ -6,3 +6,64 @@ select (name,escape_attempts) from animals where weight_kg > 10.5;
...
+
+begin;
+delete from animals;
+select * from animals;
+rollback;
```
Comment (bobb-Rob) [inline queries.sql:28]: - [x] After the rollback on line 26, you should add a query to verify that the change was undone.

File: queries.sql [hunk 5]
```diff
@@ -6,3 +6,64 @@ select (name,escape_attempts) from animals where weight_kg > 10.5;
...
+-- What is the minimum and maximum weight of each type of animal?
+-- Answer: Pokemon (min:11, max:17), Digimon (min:5.7, max:45)
+-- Query:
+select species,min(weight_kg),max(weight_kg) from animals group by species;
+
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+-- Answer: Pokemon(3), Digimon(Null)
+-- Query:
+select species,avg(escape_attempts) from animals where date_part('year',date_of_birth) between 1990 and 2000 group by species;
```
Comment (bobb-Rob) [inline queries.sql:73]: - [x] On your pull request, the last screenshot attached appears to be the wrong one. Kindly confirm this and ensure you attached the correct one.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @wickathou, 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of GitHub flow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: data.sql (hunk 1 above)
Comment (MaganaAsiati) [inline data.sql]: Kindly make sure that everything that creates or changes the table structure should be saved in the schema.sql file. please add `alter table animals add column species varchar(100)`  in the schema.sql as required by the previous reviewer [here](https://github.com/wickathou/vet-clinic-sql-practice/pull/2/files#r1157188850) :+1:

**Review by MaganaAsiati — CHANGES_REQUESTED** [review]
# STATUS: Required Changes ♻️♻️♻️ 

Hi @wickathou 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

AlonzoLeroneThisCloseGIF

## Highlights
✅ No linter error
✅ README is professional

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: data.sql (hunk 1 above)
Comment (wickathou) [reply data.sql]: I noticed I forgot to push the changes :sweat_smile:

**Review by omarsalem7 — APPROVED** [review]
Hi @wickathou 😊, 
# STATUS: APPROVED! 🥇 💯 

giphy

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 🚀 
Congratulations! 🎉 

## Highlights
✅ Project has met all the technical aspects you needed for this project. Keep rocking! 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**